File: frontend/src/app/services/auth.service.interface.ts
import {Observable} from "rxjs";
import {IUserModel} from "../models/user.model.interface";

export interface IAuthCapabilities {
  googleLoginEnabled: boolean;
  passwordRecoveryEmailEnabled: boolean;
  localPreviewEnabled: boolean;
}

export interface IAuthService {
  getCapabilities(): Observable<IAuthCapabilities>;
	openLocalPreview(): Observable<void>;
  login(username: string, password: string): Observable<Object>;
  register(email: string, password: string): Observable<IUserModel>;
  requestPasswordReset(email: string): Observable<void>;
  confirmPasswordReset(token: string, newPassword: string): Observable<void>;
  logout(): Observable<void>;
  loggedIn(): Observable<boolean>;
}

File: idp/internal/app/authentication/auth_service_interface.go
package authentication

import (
	"automation-hub-idp/internal/app/dto"
	"context"
	"github.com/google/uuid"
	"time"
)

type IService interface {
	Capabilities() dto.AuthCapabilities
	Register(userDTO dto.UserDTO) (*dto.UserResponse, error)
	Login(email, password string) (*dto.TokenDetails, error)
	// GoogleAuthURL returns the Google consent URL for "Sign in with Google".
	GoogleAuthURL() (string, error)
	// LoginWithGoogle completes the Google flow and returns a HAI session,
	// creating the user on first sign-in.
	LoginWithGoogle(ctx context.Context, code, state string) (*dto.TokenDetails, error)
	// LocalPreviewLogin is an explicitly enabled, local-only owner session for
	// a single-user installation. It is not a general authentication bypass.
	LocalPreviewLogin() (*dto.TokenDetails, error)
	Logout(accessToken string) error
	RefreshToken(refreshToken string) (*dto.TokenDetails, error)
	IsUserAuthenticated(accessToken string) (bool, error)
	RequestPasswordReset(email string) (string, time.Time, error)
	ConfirmPasswordReset(token, newPassword string) error
	ChangePassword(accessToken string, newPassword string) error
	GetIdFromToken(accessToken string) (uuid.UUID, error)
	GetSessionFromToken(accessToken string) (*dto.AuthSession, error)
}
